mediathread/assetmgr/views.py

```diff
@@ -324,6 +328,62 @@ def post(self, request):
             return HttpResponseRedirect(asset_url)
 
 
+class UploadedAssetCreateView(LoggedInCourseMixin, View):
+    """
+    View for creating an Asset via an uploaded media object.
+    """
+    http_method_names = ['post']
+
+    def dispatch(self, request, *args, **kwargs):
+        r = super().dispatch(request, *args, **kwargs)
+
+        if not (request.user.is_staff or
+                request.course.is_faculty(request.user)):
+            raise PermissionDenied
+
+        return r
+
+    def post(self, request, *args, **kwargs):
+        if (not request.POST.get('title')) or \
+           (not request.POST.get('url')):
+            return HttpResponseBadRequest(
+                'Title and URL are required to make an asset.')
+
+        title = request.POST.get('title').strip()
+        url = request.POST.get('url')
+
+        asset = Asset.objects.create(
```

Comment:
Check for the "Upload As" parameter and ensure the author is being set as specified.